odoo/service/server.py

```diff
@@ -856,17 +849,22 @@ def start(self):
         signal.signal(signal.SIGCHLD, signal.SIG_DFL)
         signal.set_wakeup_fd(self.wakeup_fd_w)
 
+        set_limit_memory_hard()
+
+        # SIGXCPU (exceeded CPU time) signal handler will raise an exception.
+        def time_expired(n, stack):
+            _logger.info('Worker (%d) CPU time limit (%s) reached.', self.pid, config['limit_time_cpu'])
+            # blow up compltely, don't wait for request end
+            raise Exception('CPU time limit exceeded.')
+        signal.signal(signal.SIGXCPU, time_expired)
+
     def stop(self):
         pass
 
     def run(self):
         try:
             self.start()
-            while self.alive:
-                self.process_limit()
-                self.multi.pipe_ping(self.watchdog_pipe)
-                self.sleep()
-                self.process_work()
+            run_async('Worker %s (%s) work' % (type(self).__name__, self.pid), self._runloop)
```

Comment:
It's the only line needed, so i would just do it manually:
threading.Thread(target=_runloop);t.setDaemon(True);t.start()